crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -827,8 +827,10 @@ mod test {
 
     // From https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html
     #[test]
-    // allow: we want explicit assertions here to match the documented expected behavior.
-    #[allow(clippy::nonminimal_bool)]
+    #[expect(
+        clippy::nonminimal_bool,
+        reason = "we want explicit assertions here to match the documented expected behavior"
```

Comment:
We should have a followup to fix these. The explicit assertions could be done in a way that makes it easier for the reader.

Reply:
https://github.com/astriaorg/astria/issues/1583